src/main/java/net/dv8tion/jda/internal/entities/GuildThreadImpl.java

```diff
@@ -0,0 +1,241 @@
+package net.dv8tion.jda.internal.entities;
+
+import net.dv8tion.jda.api.entities.*;
+import net.dv8tion.jda.api.entities.GuildThread;
+import net.dv8tion.jda.api.requests.restaction.ChannelAction;
+import org.jetbrains.annotations.NotNull;
+import sun.reflect.generics.reflectiveObjects.NotImplementedException;
+
+import java.time.Instant;
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+import java.util.List;
+
+public class GuildThreadImpl extends AbstractChannelImpl<GuildThread, GuildThreadImpl> implements GuildThread
+{
+    //TODO this same pattern is used in MemberImpl. Might want to consider centralizing? dunno.
+    private static final ZoneOffset OFFSET = ZoneOffset.of("+00:00");
+
+    //TODO | If / When we introduce PublicThread/PrivateThread/NewsThread then we wont need this field
+    private final ChannelType channelType;
+
+    private GuildThreadMember selfThreadMember;
+    private AutoArchiveDuration autoArchiveDuration;
+    private boolean isArchived;
+    private long parentChannelId;
+    private long archivingMemberId;
+    private long archiveTimestamp;
+    private long ownerId;
+    private long lastMessageId;
+    private int messageCount;
+    private int memberCount;
+
+    public GuildThreadImpl(long id, GuildImpl guild, ChannelType channelType)
+    {
+        super(id, guild);
+        this.channelType = channelType;
+    }
+
+    @Override
+    public long getLatestMessageIdLong()
+    {
+        return lastMessageId;
+    }
+
+    @Override
+    public boolean hasLatestMessage()
+    {
+        return lastMessageId != 0;
+    }
+
+    @NotNull
+    @Override
+    public ChannelType getType()
+    {
+        return channelType;
+    }
+
+    @NotNull
+    @Override
+    public List<Member> getMembers()
+    {
+        throw new NotImplementedException();
+    }
+
+    @Override
+    public int getPosition()
+    {
+        throw new UnsupportedOperationException("Thread do not have a concept of position");
```

Comment:
How does the client order them? Can we not just use the same ordering?

Reply:
It is either based on join date or creation date. Will need to test, but yeah, it does have an ordering